File: X&0.c
#include <stdio.h>

#define SIZE 3

void printBoard(char board[SIZE][SIZE]) {
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            printf("%c ", board[i][j]);
        }
        printf("\n");
    }
}

void initializeBoard(char board[SIZE][SIZE]) {
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            board[i][j] = ' ';
        }
    }
}

void playerMove(char board[SIZE][SIZE], char player) {
    int row, col;
    while (1) {
        printf("Player %c, Enter row and column (0, 1, 2): ", player);
        scanf("%d %d", &row, &col);

        // Joyni tekshirish
        if (row >= 0 && row < SIZE && col >= 0 && col < SIZE && board[row][col] == ' ') {
            board[row][col] = player;
            break;
        } else {
            printf("Invalid move! Try again.\n");
        }
    }
}

int checkWinner(char board[SIZE][SIZE]) {
    for (int i = 0; i < SIZE; i++) {
        if (board[i][0] == board[i][1] && board[i][1] == board[i][2] && board[i][0] != ' ') {
            return (board[i][0] == 'X') ? 1 : 2;
        }
        if (board[0][i] == board[1][i] && board[1][i] == board[2][i] && board[0][i] != ' ') {
            return (board[0][i] == 'X') ? 1 : 2;
        }
    }

    if (board[0][0] == board[1][1] && board[1][1] == board[2][2] && board[0][0] != ' ') {
        return (board[0][0] == 'X') ? 1 : 2;
    }
    if (board[0][2] == board[1][1] && board[1][1] == board[2][0] && board[0][2] != ' ') {
        return (board[0][2] == 'X') ? 1 : 2;
    }

    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            if (board[i][j] == ' ') {
                return 0;
            }
        }
    }

    return 3;
}

int main() {
    char board[SIZE][SIZE];
    initializeBoard(board);

    int winner = 0;
    int currentPlayer = 1;
    while (winner == 0) {
        printBoard(board);

        playerMove(board, currentPlayer == 1 ? 'X' : 'O');

        winner = checkWinner(board);

 
        currentPlayer = (currentPlayer == 1) ? 2 : 1;
    }

	printBoard(board);
    if (winner == 1) {
        printf("Player 1 (X) wins!\n");
    } else if (winner == 2) {
        printf("Player 2 (O) wins!\n");
    } else {
        printf("It's a draw!\n");
    }

    return 0;
}

File: X&0.py
def print_board(board):
    print()
    for row in board:
        print(" | ".join(row))
        print("-" * 5)
    print()


def check_winner(board, player):
    for row in board:
        if all(cell == player for cell in row):
            return True

    for col in range(3):
        if all(board[row][col] == player for row in range(3)):
            return True

    if all(board[i][i] == player for i in range(3)):
        return True
    if all(board[i][2 - i] == player for i in range(3)):
        return True

    return False


def is_full(board):
    return all(cell != " " for row in board for cell in row)


def play_game():
    board = [[" " for _ in range(3)] for _ in range(3)]
    current_player = "X"

    print("X va O o‘yini!")
    print_board(board)

    while True:
        try:
            row = int(input(f"{current_player} - qatorni kiriting (0-2): "))
            col = int(input(f"{current_player} - ustunni kiriting (0-2): "))
        except ValueError:
            print("Faqat son kiriting!")
            continue

        if row not in range(3) or col not in range(3):
            print("Xatolik: qator va ustun 0 dan 2 gacha bo‘lishi kerak!")
            continue

        if board[row][col] != " ":
            print("Bu joy allaqachon band!")
            continue

        board[row][col] = current_player
        print_board(board)

        if check_winner(board, current_player):
            print(f"{current_player} yutdi! Tabriklaymiz! 🎉")
            break

        if is_full(board):
            print("Durrang!")
            break

        current_player = "O" if current_player == "X" else "X"


play_game()

